entities/weapons/keys/shared.lua

```diff
@@ -137,7 +137,7 @@ end
 
 function SWEP:Reload()
     local trace = self:GetOwner():GetEyeTrace()
-    if not IsValid(trace.Entity) or (IsValid(trace.Entity) and ((not trace.Entity:isDoor() and not trace.Entity:IsVehicle()) or self.Owner:EyePos():DistToSqr(trace.HitPos) > 40000)) then
+    if not IsValid(trace.Entity) or (((not trace.Entity:isDoor() and not trace.Entity:IsVehicle()) or self.Owner:EyePos():DistToSqr(trace.HitPos) > 40000)) then
```

Comment:
But the check is in bracktes, while the IsValid check is not.
The IsValid checks then runs after all the other checks

Reply:
@mcNuggets1 That's fine, the if-statement is short circuited if the entity is invalid - the second set of statements will only be reached with a valid entity, making the second IsValid useless.

Reply:
me idiot. need to get more rest nowadays, haha.